ocs_ci/ocs/resources/objectbucket.py

```diff
@@ -215,6 +224,8 @@ def delete(self, verify=True):
             verify = True
         if verify:
             self.verify_deletion()
+        if original_context:
```

Comment:
Thanks, it would be good to use those functions more. I am not sure if it would work in this case. There is a variable `self.cluster_context` that is used as a context of instance of the class and there can be multiple instances on multiple clusters. I am not sure if the decorator would keep that recorded correctly but for function calls it makes sense.